### PR TITLE
feat(ci): add nightly builds, rename zips

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

Part of https://github.com/interledger/web-monetization-extension/pull/508

## Changes proposed in this pull request

- Add nightly builds Mon-Sat 12AM UTC
- Update build script to generate correct zip names

### CODE REVIEW

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">bb54cd9d0d59bb47a012d4c12c5b89447d52bf3d</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/10451936497">Run #10451936497</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center">N/A</td></tr><tr><td align="center">Badge</td><td align="center">N/A</td></tr>
</tbody>
</table>

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -31,10 +31,10 @@ jobs:
         run: pnpm build ${{ matrix.browser }} nightly
 
       - name: Upload artifacts
-        uses: actions/upload-artifact@v3.1.3
+        uses: actions/upload-artifact@v4
         with:
           name: ${{ github.event.pull_request.number }}-${{ matrix.browser }}
-          path: dist/${{ matrix.browser }}.zip
+          path: dist/nightly-${{ matrix.browser }}-*.zip
```
Comment (sidvishnoi) [inline .github/workflows/pr-checks.yml:37]:
```suggestion
          path: dist/{{ matrix.browser }}
```

Comment (sidvishnoi) [reply .github/workflows/pr-checks.yml:37]: Done in https://github.com/interledger/web-monetization-extension/pull/518

File: .github/workflows/nightly-build.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Environment setup
+        uses: ./.github/actions/setup
+
+      - name: Build
+        id: build
+        run: |
+          pnpm build
```
Comment (sidvishnoi) [inline .github/workflows/nightly-build.yaml:30]: Just to be explicit:

```suggestion
          pnpm build --channel=nightly
```

File: .github/actions/get-built-version.cjs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// @ts-check
+/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-var-requires */
+const path = require('node:path')
+const fs = require('node:fs/promises')
+
+/** * Retrieves the manifest version from the built extension.
+    *
+    * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+    * */
```
Comment (sidvishnoi) [inline .github/actions/get-built-version.cjs:9]:
```suggestion
/**
 * Retrieves the manifest version from the built extension.
 * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
 */
```

File: .github/actions/get-built-version.cjs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+module.exports = async ({ core }) => {
+  // Retrieves the manifest version from the built extension.
```
Comment (sidvishnoi) [inline .github/actions/get-built-version.cjs:11]:
```suggestion
```

**Review by raducristianpopa — APPROVED** [review]

**Pull request merged**